src/store_list.ts

```diff
@@ -0,0 +1,94 @@
+import { ListStoresResponse } from './backend/list_stores.ts'
+import { Client, ClientOptions, getClientOptions } from './client.ts'
+import { getEnvironmentContext } from './environment.ts'
+import { DEPLOY_STORE_PREFIX, SITE_STORE_PREFIX } from './store.ts'
+import { HTTPMethod } from './types.ts'
+import { collectIterator } from './util.ts'
+
+export function listStores(options: Partial<ClientOptions> & { paginate: true }): AsyncIterable<ListStoresResponse>
+export function listStores(options?: Partial<ClientOptions> & { paginate?: false }): Promise<ListStoresResponse>
+export function listStores(
+  options: Partial<ClientOptions> & { paginate?: boolean } = {},
+): AsyncIterable<ListStoresResponse> | Promise<ListStoresResponse> {
+  const context = getEnvironmentContext()
+  const clientOptions = getClientOptions(options, context)
+  const client = new Client(clientOptions)
+  const iterator = getListIterator(client, SITE_STORE_PREFIX)
+
+  if (options.paginate) {
+    return iterator
+  }
+
+  // We can't use `async/await` here because that would make the signature
+  // incompatible with one of the overloads.
+  // eslint-disable-next-line promise/prefer-await-to-then
+  return collectIterator(iterator).then((results) =>
+    results.reduce(
+      (acc, item) => ({
+        ...acc,
+        stores: [...acc.stores, ...item.stores],
+      }),
+      { stores: [] },
+    ),
+  )
+}
+
+const formatListStoreResponse = (rawStores: string[]) =>
+  rawStores.reduce((acc, rawStore) => {
```

Comment:
nit: this could be expressed more concisely as
```js
rawStores
.filter(store => !store.startsWith(DEPLOY_STORE_PREFIX)
.map(store => store.startsWith(SITE_STORE_PREFIX) ? store.slice(SITE_STORE_PREFIX.length) : store)
```

Reply:
Done in https://github.com/netlify/blobs/pull/149/commits/b5c5bd4e6ef16fa62145a851f8503b11519764f6.